calico/common.py

```diff
@@ -435,34 +451,68 @@ def validate_endpoint(config, combined_id, endpoint):
         raise ValidationFailed(" ".join(issues))
 
 
-def validate_rules(profile_id, rules):
+def validate_tier_data(tier, data):
+    issues = []
+    if not VALID_ID_RE.match(tier):
+        issues.append("Invalid profile_id '%r'." % tier)
+
+    if not isinstance(data, dict):
+        raise ValidationFailed("Expected tier data to be a dict.")
```

Comment:
Re using  `issues` this raise is to protect the code below from trying to access a non-dict.  We'd have to put the rest of the method behind a conditional.